api/src/main/java/org/openmrs/module/attachments/obs/DefaultAttachmentHandler.java

```diff
@@ -15,7 +13,7 @@ public DefaultAttachmentHandler() {
 	}
 
 	protected void setParentComplexObsHandler() {
-		setParent(new BinaryDataHandler());
+		setParent(Context.getRegisteredComponents(BinaryDataHandler.class).get(0));
```

Comment:
In the Core 2.8.x, the exisiting handlers have become beans.

Reply:
It's a very strange construct. I would just override setParent and annotate it with @Autowired @Qualifier("binaryDataHandler") to have it injected by Spring.